subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -1786,21 +1788,50 @@ private void assertUsageIsMutable() {
         }
     }
 
-    private void logChangingUsage(String usage, boolean allowed) {
-        String msgTemplate = "Allowed usage is changing for %s, %s. Ideally, usage should be fixed upon creation.";
-        if (warnOnChangingUsage) {
-            DeprecationLogger.deprecateBehaviour(String.format(msgTemplate, getDisplayName(), describeChangingUsage(usage, allowed)))
+    private void maybeWarnOnChangingUsage(String usage, boolean current) {
+        if (!isSpecialCaseOfChangingUsage(usage, current)) {
+            String msgTemplate = "Allowed usage is changing for %s, %s. Ideally, usage should be fixed upon creation.";
```

Comment:
Now that we no longer have the info log, does it make sense to combine this template and `describeChangingUsage` into a single string template for improved cohesiveness?